### PR TITLE
Upgrade: csi-attacher from v1.2.0 to v2.1.0

### PR DESCRIPTION
**Pull request by hswong3i**
Upgrade: csi-attacher from v1.2.0 to v2.1.0

See https://github.com/kubernetes-csi/external-attacher/releases/tag/v2.1.0
See https://github.com/kubernetes-csi/external-attacher/blob/v2.1.0/CHANGELOG-2.1.md

### CODE REVIEW

**Comment by hswong3i** [comment]
@Madhu-1 breaking changes handled.

**Comment by Madhu-1** [comment]
CI still failing :(

**Comment by Madhu-1** [comment]
please squash the commits

**Comment by hswong3i** [comment]
> please squash the commits

Done

**Comment by hswong3i** [comment]
> CI still failing :(



> CI still failing :(

Recent commit fix the CI, see https://travis-ci.org/ceph/ceph-csi/builds/631821904

Now squash the commits once again.

**Comment by hswong3i** [comment]
Code rebased

**Comment by Madhu-1** [comment]
@hswong3i can you please update commit message based on #773

**Comment by hswong3i** [comment]
> @hswong3i can you please update commit message based on #773

Done

**Comment by hswong3i** [comment]
> Minimum kubernetes version for v2.0.0 of the attacher is v1.14 as per the [release](https://github.com/kubernetes-csi/external-attacher/releases/tag/v2.0.0) notes.
> 
> I would hence suggest we not change the v1.13 manifests in this change and leave it as 1.2.0.

Therefore partly rollback to csi-attacher:v1.2.1 for kubernetes v1.13?

**Comment by hswong3i** [comment]
Rebased

File: scripts/minikube.sh [hunk 1]
```diff
@@ -117,7 +117,7 @@ cephcsi)
     ;;
 k8s-sidecar)
     echo "copying the kubernetes sidecar images"
-    copy_image_to_cluster "${K8S_IMAGE_REPO}"/csi-attacher:v1.2.0 "${K8S_IMAGE_REPO}"/csi-attacher:v1.2.0
+    copy_image_to_cluster "${K8S_IMAGE_REPO}"/csi-attacher:v2.0.0 "${K8S_IMAGE_REPO}"/csi-attacher:v2.0.0
```
Comment (ShyamsundarR) [inline scripts/minikube.sh]: (nit) we possibly also need to copy the 1.2.1 image to the cluster, as v1.13 kubernetes tests would use the same.

Comment (hswong3i) [reply scripts/minikube.sh]: Done and rebased

**Comment by Madhu-1** [comment]
attacher 2.1 is released can this PR be updated https://github.com/kubernetes-csi/external-attacher/blob/v2.1.0/CHANGELOG-2.1.md

**Comment by hswong3i** [comment]
> attacher 2.1 is released can this PR be updated https://github.com/kubernetes-csi/external-attacher/blob/v2.1.0/CHANGELOG-2.1.md

Done

**Pull request merged**